src/Middleware/Diagnostics.EntityFrameworkCore/src/DatabaseDeveloperPageExceptionFilter.cs

```diff
@@ -38,7 +38,8 @@ public async Task HandleExceptionAsync(ErrorContext errorContext, Func<ErrorCont
             {
                 // Look for DbContext classes registered in the service provider
                 var registeredContexts = errorContext.HttpContext.RequestServices.GetServices<DbContextOptions>()
-                    .Select(o => o.ContextType);
+                    .Select(o => o.ContextType)
+                    .Distinct();
```

Comment:
Since the PR is still building, could we make a small note about the workaround:

```suggestion
                    .Distinct(); // Workaround for https://github.com/dotnet/efcore/issues/22341
```